.github/workflows/build-and-test.yml

```diff
@@ -8,120 +8,8 @@ on:
       - "*"
 
 jobs:
-  setup_for_spack:
-    name: Run setup.py phases needed by spack
-    needs: [setup_test]
-    runs-on: ubuntu-latest
-    container:
-      image: precice/precice:develop
-      options: --user root
-    steps:
-      - name: Checkout Repository
-        uses: actions/checkout@v2
-      - name: Install & upgrade pip3
-        run: |
-          apt-get -yy update
-          apt-get install -y python3-pip pkg-config
-          rm -rf /var/lib/apt/lists/*
-          pip3 install --upgrade --user pip
-      - name: Install dependencies
-        run: |
-          pip3 install --user toml
-          python3 -c 'import toml; c = toml.load("pyproject.toml"); print("\n".join(c["build-system"]["requires"]))' | pip3 install -r /dev/stdin
-      - name: Run setup.py phases needed by spack
-        run: |
-          python3 setup.py install_lib
-          python3 setup.py build_ext
-
-  setup_install:
-    name: Run setup install
-    runs-on: ubuntu-latest
-    container:
-      image: precice/precice:develop
-      options: --user root
-    steps:
-      - name: Checkout Repository
-        uses: actions/checkout@v2
-      - name: Install pip3, pkgconfig and upgrade pip3
-        run: |
-          apt-get -yy update
-          apt-get install -y python3-pip pkg-config
-          rm -rf /var/lib/apt/lists/*
-          pip3 install --upgrade --user pip
-      - name: Install dependencies
-        run: |
-          pip3 install --user toml
-          python3 -c 'import toml; c = toml.load("pyproject.toml"); print("\n".join(c["build-system"]["requires"]))' | pip3 install -r /dev/stdin
-      - name: Run setup install
-        run:  python3 setup.py install --user
-      - name: Test install
-        run: |
-          export LD_LIBRARY_PATH=/usr/local/lib:$LD_LIBRARY_PATH
-          python3 -c "import precice"
```

Comment:
The standard installation using `python3 setup.py install --user` we should still keep (maybe update to `pip install .`?

Reply:
Invoking setup.py with arguments is deprecated. There is no benefit in keeping this.
https://blog.ganssle.io/articles/2021/10/setup-py-deprecated.html

Reply:
But we could do `pip install` instead to regularly test the installation procedure, no?

Reply:
Restored